bot/moderation/moderation.py

```diff
@@ -768,6 +792,34 @@ async def modmail_reaction_remove(self, payload: discord.RawReactionActionEvent)
                 new_embed = await self.change_modmail_status(modmail, payload.emoji.name, False)
                 await modmail.edit(embed=new_embed)
 
+    async def check_warnings(self, ctx: commands.Context, user: discord.Member):
+        """Method which checks the amount of warnings a user has and punishes him if necessary.
+
+        The individual punishments and the amount of warnings needed to trigger it are stored in a dictionary and can
+        therefore be easily modified or expanded. If a specific limit has been reached the bot simply invokes one of his
+        own mod commands.
+
+        Args:
+            ctx (discord.ext.commands.Context): The context in which the command was called.
+            user (discord.Member): The member which has been warned.
+        """
+        cntr_warnings = len(self._db_connector.get_member_warnings(user.id))
+        punishments = {
+            constants.LIMIT_WARNINGS_LVL_1:      ("tempmute", "1 week"),
+            constants.LIMIT_WARNINGS_LVL_2:      ("tempban", "2 weeks"),
+            constants.LIMIT_WARNINGS_LVL_3:      ("ban", None)
+        }
+
+        if cntr_warnings in punishments:
+            punishment = punishments[cntr_warnings]
+            reason = f"Automatisch durchgeführte Aktion aufgrund von insgesamt {cntr_warnings} Verwarnungen."
+
+            if punishment[1]:
+                await ctx.invoke(self.bot.get_command(punishment[0]), user=user, duration=punishment[1], reason=reason,
+                                 bot_activated=True)
+            else:
+                await ctx.invoke(self.bot.get_command(punishment[0]), user=user, reason=reason, bot_activated=True)
```

Comment:
Do you actually need this if/else?
If the duration is the only difference, I assume the default value if it is not passed is None anyway.

Reply:
Yeah but I think this would still result in a TypeError because of an unexpected keyword argument. The else is for methods which don't take a duration because they're permanent.